### PR TITLE
Add hostonly mac devicelab test.

### PR DESCRIPTION
**Pull request by godofredoc**
Bug:
  https://github.com/flutter/flutter/issues/65223

### CODE REVIEW

File: config/devicelab_config.star [hunk 1]
```diff
@@ -70,8 +70,458 @@ def devicelab_prod_config(branch, version, ref):
         )
 
     # Defines framework prod builders
+    common.mac_prod_builder(
```
Comment (keyonghan) [inline config/devicelab_config.star:73]: Add a doc about mac builders?

File: config/devicelab_config.star [hunk 2]
```diff
@@ -281,6 +731,440 @@ def devicelab_try_config():
     )
 
     # Defines devicelab try builders
+    common.mac_try_builder(
```
Comment (keyonghan) [inline config/devicelab_config.star:734]: ditto.

**Review by keyonghan — APPROVED** [review]
LGTM with nits.

File: config/devicelab_config.star (hunk 1 above)
Comment (godofredoc) [reply config/devicelab_config.star:75]: done

File: config/devicelab_config.star (hunk 2 above)
Comment (godofredoc) [reply config/devicelab_config.star:736]: done

**Pull request merged**